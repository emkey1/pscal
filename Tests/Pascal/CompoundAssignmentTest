program CompoundAssignmentTest;

var
  total, aces: integer;

function NextValue(base: integer): integer;
begin
  NextValue := base + 2;
end;

begin
  total := 5;
  total += 10;
  total -= 4;

  aces := 2;
  aces -= 1;

  writeln('After first round:');
  writeln('total=', total);
  writeln('aces=', aces);

  total += NextValue(3);
  aces += total - 12;

  writeln('After expression round:');
  writeln('total=', total);
  writeln('aces=', aces);

  total -= aces;

  writeln('After final subtraction:');
  writeln('total=', total);
  writeln('aces=', aces);
end.
