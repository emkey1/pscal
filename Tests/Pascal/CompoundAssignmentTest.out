After first round:
total=11
aces=1
After expression round:
total=16
aces=5
After final subtraction:
total=11
aces=5
